### DISCUSSION TITLE
In the lock file, what are the hashes after a version number of a dependency?

### DISCUSSION THREAD

**Discussion by supalarry** [pnpm/pnpm#5225] [Q&A] [answered]
Hi there,

Why do some dependencies have `_string` appended to their version in the pnpm-lock.yaml?

```
    dependencies:
      micro-cors: 0.1.1
      next: 12.1.0_sfoxds7t5ydpegc3knd667wn6m
```

1. What does `_sfoxds7t5ydpegc3knd667wn6m` mean?
2. When does it change?

Thank you and have a nice day! : )

**Comment by zkochan** [comment, marked as answer]
It is a hash of the resolved peer dependency names and versions: https://pnpm.io/how-peers-are-resolved

**Reply by supalarry** [reply]
Thank you. I was asking because we have a Next app deployed using Vercel. The hash change leads to Vercel not being able to re-use cache. In the screenshot below the highlighted pink part had changed in the `pnpm-lock.yaml` file, but Vercel still had old one. 

However, why would the highlighted pink part need to change? We have not changed our Next version (12.1.2) which is a direct dependency for our app. Notably, we had `auto-install-peers = true` which I set to false and now will see if it helps.

<img width="910" alt="Screenshot 2022-08-17 at 15 13 10" src="https://user-images.githubusercontent.com/42170848/185143111-2fdc0c2d-7e66-4f33-99a9-01d25bc523fa.png">

**Reply by zkochan** [reply]
you don't have to change next to get the hash changed. The hash changes when one or many of the peer dependencies used by next are changed. The peers are resolved from parent packages.

**Reply by supalarry** [reply]
Thanks, that answers my questions!
